### PR TITLE
HPCC-12942 Process standard query responses correctly

### PR DESCRIPTION
**Pull request by rpastrana**
- XmlWriter should allow fully qualified ns (don't append xmlns automatically)
- Query responses should be named appropriately
- Remove redundant response processing code

Signed-off-by: rpastrana rodrigo.pastrana@lexisnexis.com

### CODE REVIEW

**Comment by rpastrana** [comment]
@afishbeck please review. In particular please consider the change to the xmlwriter ns logic. Also, left commented out code in order to show the actual change. I will remove after your review. Thanks.

File: common/thorhelper/thorxmlwrite.cpp [hunk 1]
```diff
@@ -246,7 +246,10 @@ void CommonXmlWriter::outputXmlns(const char *name, const char *uri)
 {
     StringBuffer fieldname;
     if (!streq(name, "xmlns"))
-        fieldname.append("xmlns:");
```
Comment (afishbeck) [inline common/thorhelper/thorxmlwrite.cpp]: This change isn't necessary, or actually correct.  Should probably be an exception if there is a colon in the xmlns.. see comment below.

File: esp/esdllib/esdl_transformer2.cpp [hunk 2]
```diff
@@ -1317,7 +1317,7 @@ void Esdl2Response::process(Esdl2TransformerContext &ctx, const char *out_name,
 
                     if (ctx.schemaLocation.length() > 0 )
                     {
-                        ctx.writer->outputXmlns("xmlns:xsi", "http://www.w3.org/2001/XMLSchema-instance");
+                        ctx.writer->outputXmlns("xsi", "http://www.w3.org/2001/XMLSchema-instance");
                         ctx.writer->outputXmlns("xsi:schemaLocation", ctx.schemaLocation.str());
```
Comment (afishbeck) [inline esp/esdllib/esdl_transformer2.cpp]: This is not a namespace, and shouldn't call outputXmlns.  This is an attribute that uses the "xsi"  namespace.

**Comment by rpastrana** [comment]
@afishbeck please review latest change.

**Comment by afishbeck** [comment]
@rpastrana looks fine.

**Comment by rpastrana** [comment]
@richardkchapman please merge this PR, thanks.

File: common/thorhelper/thorxmlwrite.cpp (hunk 1 above)
Comment (richardkchapman) [inline common/thorhelper/thorxmlwrite.cpp]: Has this comment been addressed?

Comment (rpastrana) [inline common/thorhelper/thorxmlwrite.cpp]: Yes, I was misusing this method. My original change to this method is not necessary.

Comment (afishbeck) [inline common/thorhelper/thorxmlwrite.cpp]: looks like an extra tab is still added.

File: esp/esdllib/esdl_transformer2.cpp [hunk 3]
```diff
@@ -1817,7 +1817,30 @@ void Esdl2Transformer::processHPCCResult(IEspContext &ctx, IEsdlDefMethod &mthde
                 {
                     Esdl2Response *resp = dynamic_cast<Esdl2Response *>(root);
                     if (resp)
-                        resp->processChildNamedResponse(tctx, restype);
+                        resp->process(tctx, restype);//resp->processChildNamedResponse(tctx, restype);
+                }
+            }*/
```
Comment (afishbeck) [inline esp/esdllib/esdl_transformer2.cpp]: Also you said you'd remove commented out code after my review :)

**Comment by rpastrana** [comment]
Both issues addressed.

**Pull request merged**